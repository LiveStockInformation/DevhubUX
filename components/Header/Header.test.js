import * as React from 'react'
import { render, cleanup, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom'
import { Header } from '.'

const useRouter = jest.spyOn(require('next/router'), 'useRouter')
export function mockNextUseRouter (pathname) {
  useRouter.mockImplementation(() => ({
    route: '',
    basePath: '',
    pathname,
    query: {},
    asPath: '',
    push: async () => true,
    replace: async () => true,
    reload: () => null,
    back: () => null,
    prefetch: async () => undefined,
    beforePopState: () => null,
    isFallback: false,
    events: {
      on: () => null,
      off: () => null,
      emit: () => null
    }
  }))
}

afterEach(cleanup)

describe('Header Component', () => {
  it('renders without crashing', () => {
    mockNextUseRouter('/')
    const { asFragment } = render(<Header />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders an active class on first navigation link', () => {
    mockNextUseRouter('/')
    const { container } = render(<Header />)
    const navigationItem = container.querySelector(
      '.header__navigation-item:first-child'
    )
    expect(navigationItem).toHaveClass('header__navigation-item--active')
  })
  it('toggles mobile navigation', () => {
    mockNextUseRouter('/')
    const { container } = render(<Header />)
    const toggleButton = container.querySelector(
      '.header__toggle-button'
    )
    fireEvent.click(toggleButton)
    const headerNavigation = container.querySelector('.header__navigation')
    expect(headerNavigation).toHaveClass('header__navigation--open')
  })
})
