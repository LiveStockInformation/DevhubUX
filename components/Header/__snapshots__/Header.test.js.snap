// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component renders without crashing 1`] = `
<DocumentFragment>
  <header
    class="header"
  >
    <div
      class="phase-banner"
    >
      <p
        class="phase-banner__content"
      >
        <span
          class="phase-banner__text"
        >
          The Sheep Service went live 28th March 2022, Cattle Service coming soon.
        </span>
      </p>
    </div>
    <div
      class="header__content"
    >
      <a
        class="header__skip-link"
        href="#main-content"
      >
        Skip to main content
      </a>
      <div
        class="header__ls-logo"
      >
        <a
          class="header__ls-logo-link"
          href="/"
        >
          <span
            class="vis-hide"
          >
            Livestock Developer Hub
          </span>
        </a>
      </div>
      <div
        class="header__title"
      >
        <button
          aria-controls="navigation"
          aria-expanded="false"
          aria-label="Show or hide main menu"
          class="header__toggle-button "
          type="button"
        >
          <span
            class="vis-hide"
          >
            Menu
          </span>
        </button>
      </div>
    </div>
    <nav
      aria-label="main-navigation"
      class="header__navigation "
    >
      <div
        class="header__navigation-content"
      >
        <ul
          aria-label="Top Level Navigation"
          class="header__navigation-list"
          id="navigation"
        >
          <li
            class="header__navigation-item header__navigation-item--active"
          >
            <a
              class="ls-link header__link"
              href="/"
            >
              Home
            </a>
          </li>
          <li
            class="header__navigation-item "
          >
            <a
              class="ls-link header__link"
              href="/documentation/using-the-developer-hub"
            >
              Overview
            </a>
          </li>
          <li
            class="header__navigation-item "
          >
            <a
              class="ls-link header__link"
              href="/services"
            >
              LI Services
            </a>
          </li>
          <li
            class="header__navigation-item "
          >
            <a
              class="ls-link header__link"
              href="/apis"
            >
              APIs
            </a>
          </li>
          <li
            class="header__navigation-item "
          >
            <a
              class="ls-link header__link"
              href="/my-apps-and-data"
            >
              My Apps & Data
            </a>
          </li>
          <li
            class="header__navigation-item "
          >
            <a
              class="ls-link header__link"
              href="/support"
            >
              Support
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </header>
</DocumentFragment>
`;
